ios/scaffold/Plugins/DoubleIconsPlugin.swift

```diff
@@ -0,0 +1,122 @@
+            }
+            /////////////////////////////////////////////////////////////
+        }
+
+        self.addRpcMethodShim("setRightIcon") { params, respond in
+            ////////// This will be autogenerated at some point //////////
+            if let address: MessageAddress = MethodShimUtils.getArg(params, key: "address", respond: respond) {
+                self.setRightIcon(address, respond: respond)
+            }
+            /////////////////////////////////////////////////////////////
+        }
+
+        viewController.view.frame = CGRectMake(0, 0, 88.0, 44.0)
+    }
+
+    // @RpcMethod
+    func setLeftIcon(address: MessageAddress, respond: RpcMethodCallback) {
+        if let plugin: ViewPlugin = pluginResolver.pluginInstanceByAddress(address, respond: respond) {
+            leftIconViewController = plugin.viewController
+            setIcon(plugin, triggerFunctionName: "tapLeftButton:", visualConstraintFormat: "H:|[view]")
+        }
+    }
+
+    // @RpcMethod
+    func setRightIcon(address: MessageAddress, respond: RpcMethodCallback) {
+        if let plugin: ViewPlugin = pluginResolver.pluginInstanceByAddress(address, respond: respond) {
+            rightIconViewController = plugin.viewController
+            setIcon(plugin, triggerFunctionName: "tapRightButton:", visualConstraintFormat: "H:[view]|")
+        }
+    }
+
+    private func setIcon(plugin: ViewPlugin, triggerFunctionName: String, visualConstraintFormat: String) {
+        let pluginViewController = plugin.viewController
+
+        pluginViewController.view.userInteractionEnabled = false
+        let pluginView = wrapInUIControl(pluginViewController.view, triggerFunctionName: triggerFunctionName)
+
+        // these two methods must be called together
+        viewController.addChildViewController(pluginViewController)
+        pluginViewController.didMoveToParentViewController(viewController)
+
+        viewController.view.addSubview(pluginView)
+
+        let widthContraint = NSLayoutConstraint(item: pluginView, attribute: NSLayoutAttribute.Width, relatedBy: NSLayoutRelation.Equal, toItem: viewController.view, attribute: NSLayoutAttribute.Width, multiplier: 0.5, constant: 0)
+        let heightContraint = NSLayoutConstraint(item: pluginView, attribute: NSLayoutAttribute.Height, relatedBy: NSLayoutRelation.Equal, toItem: viewController.view, attribute: NSLayoutAttribute.Height, multiplier: 1, constant: 0)
+
+        viewController.view.addConstraint(widthContraint)
+        viewController.view.addConstraint(heightContraint)
+
+        viewController.view.addConstraints(NSLayoutConstraint.constraintsWithVisualFormat(visualConstraintFormat, options: [], metrics: nil, views: ["view":pluginView]))
+    }
+
+    private func wrapInUIControl(view: UIView, triggerFunctionName: String) -> UIControl {
+        let subViewSize = view.frame.size
+        let viewRect = CGRectMake(CGFloat(0), CGFloat(0), subViewSize.width, subViewSize.height)
```

Comment:
In Swift you can just use a CGRect constructor `CGRect(x:y:width:height:)` instead of `CGRectMake`. This will help to make it a bit more obvious what the 4 values we're passing in are.